src/html5-qrcode-scanner.ts

```diff
@@ -812,26 +808,26 @@ export class Html5QrcodeScanner {
             cameraActionStartButton.style.opacity = "0.5";
             // Swap link is available only when both scan types are required.
             if (this.scanTypeSelector.hasMoreThanOneScanType()) {
-                $this.showHideScanTypeSwapLink(false);
+                this.showHideScanTypeSwapLink(false);
             }
-            $this.resetHeaderMessage();            
+            this.resetHeaderMessage();            
 
             // Attempt starting the camera.
             const cameraId = cameraSelectUi.getValue();
-            $this.persistedDataManager.setLastUsedCameraId(cameraId);
+            this.persistedDataManager.setLastUsedCameraId(cameraId);
 
-            $this.html5Qrcode!.start(
+            this.html5Qrcode?.start(
```

Comment:
I think at an implementation level at this stage, I'd prefer `this.html5qrcode!` to indicate that this is not optional at this stage i.e. when this click event is triggered.

If the state `html5Qrcode` state is not set -- then its due to some development error and it should be corrected during development itself rather than seeing race condition in runtime.

Reply:
According to my knowledge, the best would be to create the Html5Qrcode instance in the constructor. In our case  it's not possible. This means we have to use the non null assertion operator for this variable.